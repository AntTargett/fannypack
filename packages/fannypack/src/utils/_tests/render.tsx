import * as React from 'react';
import { render } from '@testing-library/react';
// @ts-ignore
import { Provider } from '../../Provider';

export default (Component: any, { theme = {} } = {}) => render(<Provider theme={theme}>{Component}</Provider>);
