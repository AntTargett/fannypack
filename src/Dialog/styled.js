import { palette, theme } from 'styled-tools';
import styled from '../styled';
import { Box } from '../primitives';
import Pane from '../Pane';

export const DialogContent = styled(Box)`
  padding: ${theme('fannypack.layout.spacing.small')}rem;
  max-height: 50vh;
  overflow-y: scroll;
  & {
    ${theme('fannypack.Dialog.Content.base')};
  }
`;
export const DialogHeader = styled(Box)`
  border-bottom: 1px solid ${palette('whiteDarkest')};
  padding: ${theme('fannypack.layout.spacing.small')}rem;
  width: 100%;
  & {
    ${theme('fannypack.Dialog.Header.base')};
  }
`;
export const DialogFooter = styled(Box)`
  border-top: 1px solid ${palette('whiteDarkest')};
  padding: ${theme('fannypack.layout.spacing.xsmall')}rem ${theme('fannypack.layout.spacing.xsmall')}rem;
  width: 100%;
  & {
    ${theme('fannypack.Dialog.Footer.base')};
  }
`;

export default styled(Pane)`
  border-radius: 5px;
  & {
    ${theme('fannypack.Dialog.base')};
  }
`;
