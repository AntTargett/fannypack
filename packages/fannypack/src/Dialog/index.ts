import Dialog from './Dialog';
import * as styledDialog from './styled';

export default Dialog;

export * from './Dialog';
export { default as Dialog } from './Dialog';
export * from './DialogContent';
export { default as DialogContent } from './DialogContent';
export * from './DialogClose';
export { default as DialogClose } from './DialogClose';
export * from './DialogDialog';
export { default as DialogDialog } from './DialogDialog';
export * from './DialogFooter';
export { default as DialogFooter } from './DialogFooter';
export * from './DialogHeader';
export { default as DialogHeader } from './DialogHeader';
export * from './DialogTitle';
export { default as DialogTitle } from './DialogTitle';
export * from './DialogIcon';
export { default as DialogIcon } from './DialogIcon';

export { styledDialog };
