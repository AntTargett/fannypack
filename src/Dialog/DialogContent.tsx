import * as React from 'react';
import * as PropTypes from 'prop-types';
import { BoxProps as ReakitBoxProps } from '@jmoxey/reakit/ts';

import { DialogContent as _DialogContent } from './styled';

export type LocalDialogContentProps = {
  children: React.ReactNode;
  className?: string;
};
export type DialogContentProps = LocalDialogContentProps & ReakitBoxProps;

export const DialogContent: React.FunctionComponent<LocalDialogContentProps> = ({ children, ...props }) => (
  <_DialogContent {...props}>{children}</_DialogContent>
);

DialogContent.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string
};
DialogContent.defaultProps = {
  className: undefined
};

const C: React.FunctionComponent<DialogContentProps> = DialogContent;
export default C;
