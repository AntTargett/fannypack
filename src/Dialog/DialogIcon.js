// @flow
import React from 'react';

import { DialogIcon as _DialogIcon } from './styled';

type Props = {
  className?: string
};

const DialogIcon = ({ children, ...props }: Props) => <_DialogIcon {...props} />;

DialogIcon.defaultProps = {
  className: undefined
};

export default DialogIcon;
