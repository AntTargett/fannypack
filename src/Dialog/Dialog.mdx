---
name: Dialog
route: /components/Dialog
menu: Components
---

import { Playground, PropsTable } from 'docz';
import { Box } from '../primitives/index';
import { ActionButtons } from '../Button';
import Heading from '../Heading';
import Paragraph from '../Paragraph';
import Set from '../Set';
import Dialog from './index';

# Dialog

## Import

`import { Dialog } from 'fannypack'`

## Basic Usage

<Playground>
  <Dialog title="This is a title">
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse diam ipsum, cursus id placerat congue, ultrices
    eget lectus. Duis posuere, lacus sed tristique commodo, sapien turpis mollis nunc, vestibulum consectetur lectus
    augue sit amet justo.
  </Dialog>
</Playground>

## Footer

You can render a footer for the dialog using the `footer` prop.

<Playground>
  <Dialog title="This is a title" footer={<ActionButtons/>}>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse diam ipsum, cursus id placerat congue, ultrices
    eget lectus. Duis posuere, lacus sed tristique commodo, sapien turpis mollis nunc, vestibulum consectetur lectus
    augue sit amet justo.
  </Dialog>
</Playground>

## Props

<PropsTable of={Dialog} />

## Theming

### Schema

```jsx
{
  base: string | Object,
  Header: {
    base: string | Object
  },
  Content: {
    base: string | Object
  },
  Footer: {
    base: string | Object
  }
}
```
