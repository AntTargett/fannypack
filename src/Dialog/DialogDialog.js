// @flow
import React, { type Node } from 'react';

import Dialog from './styled';

type Props = {
  a11yDescriptionId?: string,
  a11yTitleId?: string,
  border?: true | 'shadow',
  children: Node,
  className?: string
};

const DialogDialog = ({ a11yDescriptionId, a11yTitleId, border, children, ...props }: Props) => (
  <Dialog aria-describedby={a11yDescriptionId} aria-labelledby={a11yTitleId} role="dialog" border={border} {...props}>
    {children}
  </Dialog>
);

DialogDialog.defaultProps = {
  a11yDescriptionId: undefined,
  a11yTitleId: undefined,
  border: true,
  className: undefined
};

export default DialogDialog;
