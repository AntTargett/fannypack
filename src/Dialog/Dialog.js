// @flow
import React, { type Element, type Node } from 'react';

import Heading from '../Heading';
import _Dialog from './styled';
import DialogContent from './DialogContent';
import DialogHeader from './DialogHeader';
import DialogFooter from './DialogFooter';

type Props = {
  border: true | 'shadow',
  children: Node,
  className?: string,
  footer?: string | Element<any>,
  title?: string | Element<any>
};

const Dialog = ({ border, children, footer, title, ...props }: Props) => (
  <_Dialog border={border} {...props}>
    {typeof title === 'string' ? (
      <DialogHeader>
        <Heading as="h5" isSubHeading>
          {title}
        </Heading>
      </DialogHeader>
    ) : (
      <DialogHeader>title</DialogHeader>
    )}
    <DialogContent>{children}</DialogContent>
    {footer && <DialogFooter>{footer}</DialogFooter>}
  </_Dialog>
);

Dialog.Content = DialogContent;
Dialog.Header = DialogHeader;

Dialog.defaultProps = {
  border: true,
  className: undefined,
  footer: undefined,
  title: undefined
};

export default Dialog;
