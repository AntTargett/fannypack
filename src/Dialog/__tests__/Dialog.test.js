import React from 'react';
import render from '../../_utils/tests/render';
import Dialog from '../Dialog';
import 'jest-styled-components';

it('renders correctly for a default dialog', () => {
  const { container } = render(
    <Dialog title="This is a title">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse diam ipsum, cursus id placerat congue,
      ultrices eget lectus. Duis posuere, lacus sed tristique commodo, sapien turpis mollis nunc, vestibulum consectetur
      lectus augue sit amet justo.
    </Dialog>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a dialog with a footer', () => {
  const { container } = render(
    <Dialog title="This is a title" footer={<div>test</div>}>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse diam ipsum, cursus id placerat congue,
      ultrices eget lectus. Duis posuere, lacus sed tristique commodo, sapien turpis mollis nunc, vestibulum consectetur
      lectus augue sit amet justo.
    </Dialog>
  );
  expect(container.firstChild).toMatchSnapshot();
});
