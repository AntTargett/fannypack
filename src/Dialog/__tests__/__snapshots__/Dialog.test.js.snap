// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for a default dialog 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c5 {
  font-weight: 700;
  line-height: 1.2;
  margin-bottom: 1rem;
}

.heading.c5 + .sub-heading {
  margin-top: -1rem;
}

.c5:last-child {
  margin-bottom: 0;
}

h1.c5 {
  font-size: 3rem;
}

h2.c5 {
  font-size: 2.5rem;
}

h3.c5 {
  font-size: 2rem;
}

h4.c5 {
  font-size: 1.5rem;
}

h5.c5 {
  font-size: 1.25rem;
}

h6.c5 {
  font-size: 1rem;
}

.c5 {
  font-weight: 600;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-radius: 3px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  border: 1px solid #e6e6e6;
}

.c6 {
  padding: 1.25rem;
  max-height: 50vh;
  overflow-y: scroll;
}

.c4 {
  border-bottom: 1px solid #e6e6e6;
  padding: 1.25rem;
  width: 100%;
}

.c2 {
  border-radius: 5px;
}

@media (max-width:480px) {
  h1.c5 {
    font-size: 2rem;
  }
}

@media (max-width:480px) {
  h2.c5 {
    font-size: 1.8rem;
  }
}

@media (max-width:480px) {
  h3.c5 {
    font-size: 1.6rem;
  }
}

@media (max-width:480px) {
  h4.c5 {
    font-size: 1.4rem;
  }
}

@media (max-width:480px) {
  h5.c5 {
    font-size: 1.2rem;
  }
}

<div
  class="c0 c1"
>
  <div
    class="c2 c3 c1"
  >
    <div
      class="c4 c1"
    >
      <h5
        class="sub-heading c5 c1"
      >
        This is a title
      </h5>
    </div>
    <div
      class="c6 c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse diam ipsum, cursus id placerat congue, ultrices eget lectus. Duis posuere, lacus sed tristique commodo, sapien turpis mollis nunc, vestibulum consectetur lectus augue sit amet justo.
    </div>
  </div>
</div>
`;

exports[`renders correctly for a dialog with a footer 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  font-family: system-ui,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Fira Sans', 'Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  color: #435a6f;
}

.c0 *,
.c0 *::before,
.c0 *::after {
  box-sizing: inherit;
}

.c0 *:focus {
  outline: 2px solid #7467c0;
  outline-offset: 2px;
}

.c5 {
  font-weight: 700;
  line-height: 1.2;
  margin-bottom: 1rem;
}

.heading.c5 + .sub-heading {
  margin-top: -1rem;
}

.c5:last-child {
  margin-bottom: 0;
}

h1.c5 {
  font-size: 3rem;
}

h2.c5 {
  font-size: 2.5rem;
}

h3.c5 {
  font-size: 2rem;
}

h4.c5 {
  font-size: 1.5rem;
}

h5.c5 {
  font-size: 1.25rem;
}

h6.c5 {
  font-size: 1rem;
}

.c5 {
  font-weight: 600;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-radius: 3px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  border: 1px solid #e6e6e6;
}

.c6 {
  padding: 1.25rem;
  max-height: 50vh;
  overflow-y: scroll;
}

.c4 {
  border-bottom: 1px solid #e6e6e6;
  padding: 1.25rem;
  width: 100%;
}

.c7 {
  border-top: 1px solid #e6e6e6;
  padding: 1rem 1rem;
  width: 100%;
}

.c2 {
  border-radius: 5px;
}

@media (max-width:480px) {
  h1.c5 {
    font-size: 2rem;
  }
}

@media (max-width:480px) {
  h2.c5 {
    font-size: 1.8rem;
  }
}

@media (max-width:480px) {
  h3.c5 {
    font-size: 1.6rem;
  }
}

@media (max-width:480px) {
  h4.c5 {
    font-size: 1.4rem;
  }
}

@media (max-width:480px) {
  h5.c5 {
    font-size: 1.2rem;
  }
}

<div
  class="c0 c1"
>
  <div
    class="c2 c3 c1"
  >
    <div
      class="c4 c1"
    >
      <h5
        class="sub-heading c5 c1"
      >
        This is a title
      </h5>
    </div>
    <div
      class="c6 c1"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse diam ipsum, cursus id placerat congue, ultrices eget lectus. Duis posuere, lacus sed tristique commodo, sapien turpis mollis nunc, vestibulum consectetur lectus augue sit amet justo.
    </div>
    <div
      class="c7 c1"
    >
      <div>
        test
      </div>
    </div>
  </div>
</div>
`;
